### PR TITLE
Adds a @pid accessor to runner, adds an optional key called 'timeout' to server manifests

### PR DESCRIPTION
**Pull request by tnorris**
This is a dependency of an internal ticket this sprint.

timeout: will be passed to ssh if it's defined in the manifest. If not, no big.

### CODE REVIEW

**Comment by CorbinSimpsonAcquia** [comment]
Hm. I think that this does what it says on the tin.

**Comment by stacybird** [comment]
It does indeed.  I was about to add my label, but how are we managing this repo?  :-)  +1

File: lib/sof/ssh.rb [hunk 1]
```diff
@@ -10,7 +10,13 @@ class Ssh
 
   def initialize(server, echo:)
     @server = server
-    @ssh_options = { :port => server.port, :compression => false, :keys => server.keys }
+
+    @ssh_options = if server.respond_to? :timeout
```
Comment (pdrakeweb) [inline lib/sof/ssh.rb]: This could be a single line:
```@ssh_options.merge(:timeout => server.timeout) if server.respond_to?(:timeout)```

Comment (tnorris) [inline lib/sof/ssh.rb]: Sweet, thanks!

**Comment by aastaneh** [comment]
LGTM.

**Pull request merged**